conn/physic/units_test.go

```diff
@@ -745,6 +745,133 @@ func TestMaxInt64(t *testing.T) {
 	}
 }
 
+func TestSetHelper(t *testing.T) {
+	succeeds := []struct {
+		in        string
+		uintbase  prefix
+		expected  int64
+		usedChars int
+	}{
+		{"1p", pico, 1, 2},
+		{"1n", pico, 1000, 2},
+		{"1u", pico, 1000000, 2},
+		{"1µ", pico, 1000000, 3},
+		{"1m", pico, 1000000000, 2},
+		{"1k", pico, 1000000000000000, 2},
+		{"1M", pico, 1000000000000000000, 2},
+		{"1p", nano, 0, 2},
+		{"1n", nano, 1, 2},
+		{"1u", nano, 1000, 2},
+		{"1µ", nano, 1000, 3},
+		{"1m", nano, 1000000, 2},
+		{"1k", nano, 1000000000000, 2},
+		{"1M", nano, 1000000000000000, 2},
+		{"1G", nano, 1000000000000000000, 2},
+		{"1p", micro, 0, 2},
+		{"1n", micro, 0, 2},
+		{"1u", micro, 1, 2},
+		{"1µ", micro, 1, 3},
+		{"1m", micro, 1000, 2},
+		{"1k", micro, 1000000000, 2},
+		{"1M", micro, 1000000000000, 2},
+		{"1G", micro, 1000000000000000, 2},
+		{"1T", micro, 1000000000000000000, 2},
+	}
+
+	fails := []struct {
+		in     string
+		prefix prefix
+	}{
+		{"1G", pico},
+		{"1T", pico},
+		{"1T", nano},
```

Comment:
please also do one for micro. In practice I'd like it to have the exact "1 too much" value, and have the maximum value in the succeeds above. Having the limits explicitly in the unit tests is informative for the readers.